addons/sms/wizard/sms_account_sender.py

```diff
@@ -0,0 +1,25 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import re
+from odoo import api, fields, models
+from odoo.addons.sms.tools.sms_api import ERROR_MESSAGES, SmsApi
+from odoo.exceptions import ValidationError
+
+
+class SMSAccountSender(models.TransientModel):
+    _name = 'sms.account.sender'
+    _description = 'SMS Account Sender Name Wizard'
+
+    account_id = fields.Many2one('iap.account', required=True)
+    sender_name = fields.Char()
+
+    @api.constrains("sender_name")
+    def _check_sender_name(self):
+        for record in self:
+            if not re.match(r"[a-zA-Z0-9\- ]{3,11}", record.sender_name):
+                raise ValidationError("Your sender name must be between 3 and 11 characters long and only contain alphanumeric characters.")
+
+    def action_set_sender_name(self):
+        status = SmsApi(self.env)._set_sender_name(self.account_id.account_token, self.sender_name)['state']
+        if status != 'success':
+            raise ValidationError(ERROR_MESSAGES.get(status, ERROR_MESSAGES['unknown_error']))
```

Comment:
This part doesn't work properly when there is an error, the response from IAP is in this format:
```py
{
    'state': 'success' or 'error',
    'error_message': 'some_error', # only present when state is 'error'
}
```
At the moment, when there is an error, it always fallback to `unknown_error` since `error` isn't in the `ERROR_MESSAGES` dictionary.

It should be something like this instead:
```suggestion
        response = SmsApi(self.env)._set_sender_name(self.account_id.account_token, self.sender_name)
        if response['state'] != 'success':
            raise ValidationError(ERROR_MESSAGES.get(response['error_message'], ERROR_MESSAGES['unknown_error']))
```

Reply:
@jmak-odoo Actually, it might make more sense to fix this on IAP side by returning the exact error directly in the `state` key. It would be more consistent with the other routes.